solidity/src/base/MathUtil.sol

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: UNLICENSED
+// This is licensed under the Cryptographic Open Software License 1.0
+pragma solidity ^0.8.28;
+
+/// A library for math utilities.
+library MathUtil {
+    /// Computes `max(1,ceil(log_2(value)))`,
+    /// the smallest integer greater than or equal to the base 2 logarithm of a number.
+    /// If the number is less than 2, the result is 1.
+    function log2Up(uint256 value_) private pure returns (uint256 exponent_) {
+        assembly {
+            /// returns max(1,ceil(log_2(value)))
+            function log2_up(value) -> exponent {
+                if value { value := sub(value, 1) }
+                exponent := 1
+                for {} shr(exponent, value) {} { exponent := add(exponent, 1) }
+            }
+            exponent_ := log2_up(value_)
+        }
+    }
+
+    function testWeCanComputeLog2Up() public pure {
```

Comment:
Move test functions outside of the libraries e.g. to `/solidity/test/MathUtil.t.sol`

Reply:
Happy to do this, but here's some context.
This library will never actually get used. Ultimately, the Yul will get extracted and moved to a different file via a preprocessing script.

The library is entirely wrapper functionality in order to test and hold the Yul itself.

See https://github.com/spaceandtimelabs/sxt-proof-of-sql/blob/main/solidity/src/Placeholder.pre.sol and https://github.com/spaceandtimelabs/sxt-proof-of-sql/tree/main/solidity/scripts as well as #512.

So, everything in this file is not really going to be production code. So, perhaps it all belongs in `test`. On the other hand, the Yul snippets are production code.

Reply:
I see. In that case it makes sense to move it all to `test`
Otherwise, external auditors will ask the same question